### ISSUE TITLE
bug: IonRouterOutlet causes rendering pages multiple times

### ISSUE THREAD

**Issue by emil-alexandrescu** [ionic-team/ionic-framework#23501]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**
[ ] **6.x**

**Current behavior:**

Once the outlet is switched from one to another, the routes are not rendered correctly, and create new page again and again as you navigate through the pages.

**Expected behavior:**

Page should be rendered only once.

**Steps to reproduce:**

Use multiple `IonRouterOutlet` and render different one on certain condition.

It was fine on @ionic/react 5.0.7, has problem on 5.6.9.

**Related code:**

```
const authOutlet = (
  <IonRouterOutlet>
    <Route path="/login" />
    <Route component={() => <Redirect to="/login" />} />
  </IonRouterOutlet>
);
const userOutlet = (
  <IonRouterOutlet>
    <Route path="/home" />
    <Route component={() => <Redirect to="/home" />} />
  </IonRouterOutlet>
);

const App = () => (
<IonReactRouter>
  {loggedIn ? userOutlet: authOutlet}
</IonReactRouter>
);

```

**Other information:**

**Ionic info:**

```
insert the output from ionic info here
```

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you reproduce this issue in an Ionic starter app and provide a link to the repo?

**Comment by liamdebeasi** [comment]
Thanks for the issue! This issue is being closed due to the lack of a code reproduction. If this is still an issue with the latest version of Ionic, please create a new issue and ensure the template is fully filled out.

Thank you for using Ionic!
